mne/io/tag.py

```diff
@@ -203,6 +204,257 @@ def _loc_to_eeg_loc(loc):
+        elif matrix_type == FIFF.FIFFT_COMPLEX_FLOAT:
+            data = np.fromstring(read_big(fid, 4 * 2 * dims.prod()),
+                                 dtype='>f4')
+            # Note: we need the non-conjugate transpose here
+            data = (data[::2] + 1j * data[1::2])
+        elif matrix_type == FIFF.FIFFT_COMPLEX_DOUBLE:
+            data = np.fromstring(read_big(fid, 8 * 2 * dims.prod()),
+                                 dtype='>f8')
+            # Note: we need the non-conjugate transpose here
+            data = (data[::2] + 1j * data[1::2])
+        else:
+            raise Exception('Cannot handle matrix of type %d yet'
+                            % matrix_type)
+        data.shape = dims
+    elif matrix_coding in (_matrix_coding_CCS, _matrix_coding_RCS):
+        from scipy import sparse
+        # Find dimensions and return to the beginning of tag data
+        pos = fid.tell()
+        fid.seek(tag.size - 4, 1)
+        ndim = int(np.fromstring(fid.read(4), dtype='>i4'))
+        fid.seek(-(ndim + 2) * 4, 1)
+        dims = np.fromstring(fid.read(4 * (ndim + 1)), dtype='>i4')
+        if ndim != 2:
+            raise Exception('Only two-dimensional matrices are '
+                            'supported at this time')
+
+        # Back to where the data start
+        fid.seek(pos, 0)
+        nnz = int(dims[0])
+        nrow = int(dims[1])
+        ncol = int(dims[2])
+        sparse_data = np.fromstring(fid.read(4 * nnz), dtype='>f4')
+        shape = (dims[1], dims[2])
+        if matrix_coding == _matrix_coding_CCS:
+            #    CCS
+            tmp_indices = fid.read(4 * nnz)
+            sparse_indices = np.fromstring(tmp_indices, dtype='>i4')
+            tmp_ptrs = fid.read(4 * (ncol + 1))
+            sparse_ptrs = np.fromstring(tmp_ptrs, dtype='>i4')
+            if (sparse_ptrs[-1] > len(sparse_indices) or
+                    np.any(sparse_ptrs < 0)):
+                # There was a bug in MNE-C that caused some data to be
+                # stored without byte swapping
+                sparse_indices = np.concatenate(
+                    (np.fromstring(tmp_indices[:4 * (nrow + 1)], dtype='>i4'),
+                     np.fromstring(tmp_indices[4 * (nrow + 1):], dtype='<i4')))
+                sparse_ptrs = np.fromstring(tmp_ptrs, dtype='<i4')
+            data = sparse.csc_matrix((sparse_data, sparse_indices,
+                                     sparse_ptrs), shape=shape)
+        else:
+            #    RCS
+            sparse_indices = np.fromstring(fid.read(4 * nnz), dtype='>i4')
+            sparse_ptrs = np.fromstring(fid.read(4 * (nrow + 1)), dtype='>i4')
+            data = sparse.csr_matrix((sparse_data, sparse_indices,
+                                     sparse_ptrs), shape=shape)
+    else:
+        raise Exception('Cannot handle other than dense or sparse '
+                        'matrices yet')
+    return data
+
+
+def _read_simple(fid, tag, shape, rlims, dtype):
```

Comment:
FYI if you look at the final diff I did end up adding the one-line docstrings, so the bureaucracy won in the end :)

> I'm still in the dark as to what made it faster. 

Looking back in the (long) thread a bit there is this nugget:

> I timed things by using line profiling to see that a non trivial amount of time was spent triaging which writing functions to use, especially when reading a measurement info block (which we use very often). The dict access implemented here fixes that.

[I just made a gist to demonstrate the idea] (https://gist.github.com/Eric89GXL/be9052ff4cf602796599). To elaborate, a conditional with 18 options, which is what the function used to use, that gets accessed 500 times (e.g., our `test-ave.fif` has 673 tags, the vast majority of which are part of `Info`) will take ~2.5-3x longer to evaluate its conditionals than an equivalent `dict.get()`-based method. This is because `dict` uses hashing for access which can be very fast, whereas a bunch of `elif` clauses must be interpreted and thus will be slow if it's used a lot. This is what took the `read_meas_info` time in that test from 4.5 to 3.0 seconds for me. The bulk of the time will still be spent actually *reading* the data as opposed to choosing the function to read the data, but this at least cuts down on the latter overhead.